src/Murder.Editor/Data/EditorDataManager.cs

```diff
@@ -358,11 +359,17 @@ internal void SaveSettings()
                 return;
             }
 
+            FileHelper.CreateDirectoryPathIfNotExists(sourcePath);
+            if (binPath != null && !string.IsNullOrEmpty(_binResourcesDirectory))
+            {
+                FileHelper.CreateDirectoryPathIfNotExists(FileHelper.GetPath(_binResourcesDirectory));
+            }
+
             // If the source is invalid and either the source resources or binaries path have not been initialized.
-            if (!Directory.Exists(Path.GetDirectoryName(sourcePath)) &&
-                (binPath != null && (!Directory.Exists(_sourceResourcesDirectory) || !Directory.Exists(_binResourcesDirectory))))
+            string fullResourcesPath = FileHelper.GetPath(_sourceResourcesDirectory);
```

Comment:
I wonder if, instead, we should do a method that initializes the sources and binaries directories base path. I think the underlying problem is that we only attempt to create the directories after this `if`, when none of the base directories exist.

Reply:
hmmm that does sound like a better path, having to add these conditionals was why i wasn't 100% sure if this was a bug vs the setup being in an unexpected state. I'll take a swing at adding an initialize method and get a separate PR for that. I can close this one if that works for you!

Reply:
sounds good! :)